### PR TITLE
Plug scan API through fdw

### PR DESCRIPTION
**Pull request by iafuture**
For now, there are no constraints generated by FDW which should result in a full table scan for all queries. This should be able to allow scan to work end-to-end though just not in an optimal way.

I'll be adding the constraint push-down in a subsequent pr

### CODE REVIEW

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 1]
```diff
@@ -322,8 +321,8 @@ K2PgStatus PgGate_ExecCreateTable(const char *database_name,
                               bool if_not_exist,
                               bool add_primary_key,
                               const std::vector<K2PGColumnDef>& columns) {
-    elog(LOG, "PgGateAPI: PgGate_NewCreateTable %s, %s, %s", database_name, schema_name, table_name);
-    auto [status, is_pg_catalog_table, schema] = makeSChema(schema_name, columns);
+    elog(DEBUG5, "PgGateAPI: PgGate_NewCreateTable %s, %s, %s", database_name, schema_name, table_name);
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: we might want to keep the log level as LOG for now to make the debug easier. DEBUG5 is way too verbose to enable in opengauss. We could fine tune the log level after we are done with bug bashing.

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 2]
```diff
@@ -430,8 +429,8 @@ K2PgStatus PgGate_ExecCreateIndex(const char *database_name,
                               const bool skip_index_backfill,
                               bool if_not_exist,
                               const std::vector<K2PGColumnDef>& columns){
-    elog(LOG, "PgGateAPI: PgGate_NewCreateIndex %s, %s, %s", database_name, schema_name, index_name);
-    auto [status, is_pg_catalog_table, schema] = makeSChema(schema_name, columns);
+    elog(DEBUG5, "PgGateAPI: PgGate_NewCreateIndex %s, %s, %s", database_name, schema_name, index_name);
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: same here

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 3]
```diff
@@ -748,12 +747,13 @@ K2PgStatus PgGate_ExecDelete(K2PgOid database_oid,
 // SELECT ------------------------------------------------------------------------------------------
 K2PgStatus PgGate_NewSelect(K2PgOid database_oid,
                          K2PgOid table_oid,
-                         const K2PgSelectIndexParams& index_params,
+                         K2PgSelectIndexParams idxp,
                          K2PgScanHandle **handle){
-    elog(LOG, "PgGateAPI: PgGate_NewSelect %d, %d", database_oid, table_oid);
-    // TODO add to memctx
+    elog(DEBUG5, "PgGateAPI: PgGate_NewSelect %d, %d", database_oid, table_oid);
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: same here

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 2 above)
Comment (iafuture) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: fixed

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 1 above)
Comment (iafuture) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: fixed

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 3 above)
Comment (iafuture) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: fixed

**Review by johnfangAFW — APPROVED** [review]

**Pull request merged**